File: 680-valid-palindrome-ii/valid-palindrome-ii.py
class Solution(object):
    def isPalindrome(self, s, left, right):
        while(left<right):
            if(s[left]!=s[right]):
                return False
            left+=1
            right-=1
        return True

    def validPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        left = 0
        right = len(s)-1
        while left < right:
            if(s[left]!=s[right]):
                return self.isPalindrome(s,left+1,right) or self.isPalindrome(s,left,right-1)
            left+=1
            right-=1
        return True
        

File: 680-valid-palindrome-ii/valid-palindrome-ii.cpp
class Solution {
public:
    bool checkPalindrome(string str,int st,int ed) {
        while(st<ed){
            if(str[st]!=str[ed]) {
                return false;
            }
            st++;
            ed--;
        }
        return true;
    }
    bool validPalindrome(string s) {
        int st = 0;
        int ed = s.size()-1;
        while(st<ed) {
            if(s[st]!=s[ed])    return checkPalindrome(s, st+1,ed) || checkPalindrome(s,st,ed-1);
            st++;
            ed--;
        }
        return true;
    }
};